lib/src/cmaf/ham/mapper/MPDMapper.ts

```diff
@@ -1,15 +1,37 @@
+import { DashManifest } from '../../utils/dash/DashManifest.js';
 import { Manifest } from '../../utils/types/index.js';
 import { Presentation } from '../types/model/index.js';
 import { IMapper } from './IMapper.js';
-
+import { mapMpdToHam } from '../../utils/dash/formatter.js';
+import { xmlToJson, jsonToXml } from '../../utils/xml.js';
+import { mapHamToMpd } from '../../utils/dash/mpdMapper.js';
 export class MPDMapper implements IMapper {
+	//TODO : Handle SegmentTemplate and SegmentList
+
 	toHam(manifest: Manifest): Presentation[] {
-		console.log(manifest);
-		throw new Error('Not implemented');
+		let dashManifest: DashManifest | undefined;
+		xmlToJson(
+			manifest.main,
+			(result: DashManifest) => (dashManifest = result),
+		);
+
+		if (!dashManifest) {
+			return [];
+		}
+
+		const presentation = mapMpdToHam(dashManifest);
+		return [presentation];
 	}
 
 	toManifest(presentation: Presentation[]): Manifest {
-		console.log(presentation);
-		throw new Error('Not implemented');
+		//TODO: Handle multiple presentations
```

Comment:
For this `toDo`, is it known what needs to be changed in the future?
It would be good to add another comment explaining (briefly) what would need to change for future work, so once we come back to this, it will be easier to remember and implement the needed changes.